modules/http/src/main/java/org/mule/module/http/internal/request/client/HttpConnectorMessageProcessorProvider.java

```diff
@@ -113,4 +114,11 @@ public void dispose()
     {
         cachedMessageProcessors.invalidateAll();
     }
+
+    @Override
+    public void stop() throws MuleException
+    {
+        cachedMessageProcessors.invalidateAll();
```

Comment:
If this is the same as `dispose` why have both?

Reply:
The issue ocurrs when the application is stopped from ARM.
In this case the dispose phase is not executed.

Reply:
Yes, but you have some other scenarios to consider:
1) The component is stopped and an undeploy occurs, you'd invalidate the cache again.
2) The component is up and running and an undeploy occurs, you'd invalidate the cache twice.